server/processors-alerts-publisher/src/main/java/org/hawkular/apm/processor/alerts/InvocationAlertsPublisherMDB.java

```diff
@@ -53,13 +55,41 @@
 
     @Override
     public void onMessage(Message message) {
-        logger.fragmentCompletionTimeReceived();
+        logger.invocationDetailsReceived();
         try {
             String data = ((TextMessage) message).getText();
-            List<CompletionTime> items = mapper.readValue(data, new TypeReference<List<CompletionTime>>() {});
-            publisher.publish(items, "FragmentCompletionTime");
+            List<NodeDetails> items = mapper.readValue(data, new TypeReference<List<NodeDetails>>() {});
+            List<Event> events = items.stream().filter(nd -> nd.isInitial())
```

Comment:
The PR is about simplifying the previous concept of "FragmentCompletionTime", which represented the overall time spent within a service, to now being the service response time (which is far easier to calculate and in most cases will be what users are interested in at this level). The 'initial' flag indicates whether the node represents the service invocation (or initial top level component). So when reporting information about invocations, these are the only nodes we are interested in.

At some point in the future we may want to allow alerts for the internal components aswell? The options we have are - (1) report the service invocations separately (as in this PR) so those triggers just interested in this info will only be triggered in these cases, or (2) report event for all nodes and then the trigger will need to distinguish the service invocations.

Difficult to decide which is best - if users are more likely to be interested in service response times, then this approach would work best. But if they want the flexibility to check duration on any internal component aswell, then may be best to just publish all to alerting engine?